src/tezos/TezosOperations.ts

```diff
@@ -337,6 +319,116 @@ export namespace TezosOperations {
         return sendOperation(network, operations, keyStore, derivationPath);
     }
 
+     /**
+       * Creates and originates a smart contract.
+       * @param {string} network  Which Tezos network to go against
+       * @param {KeyStore} keyStore   Key pair along with public key hash
+       * @param {number} amount   Initial funding amount of new account
+       * @param {string} delegate Account ID to delegate to, blank if none
+       * @param {boolean} spendable   Is account spendable?
+       * @param {boolean} delegatable Is account delegatable?
+       * @param {number} fee  Operation fee
+       * @param {string} derivationPath BIP44 Derivation Path if signed with hardware, empty if signed with software
+       * @returns {Promise<OperationResult>}  Result of the operation
+       */
+      export async function sendContractOriginationOperation(
```

Comment:
This function should be invoked by `sendOriginationOperation()` to avoid code duplication.